flink-clients/src/main/java/org/apache/flink/client/deployment/executors/AbstractSessionClusterExecutor.java

```diff
@@ -97,7 +117,26 @@ public CompletableFuture<JobClient> execute(
                                                     clusterClientProvider,
                                                     jobID,
                                                     userCodeClassloader))
-                    .whenCompleteAsync((ignored1, ignored2) -> clusterClient.close());
+                    .whenCompleteAsync(
+                            (jobClient, throwable) -> {
+                                if (throwable == null) {
+                                    RuntimeExecutionMode executionMode =
+                                            jobGraph.getJobConfiguration()
+                                                    .get(ExecutionOptions.RUNTIME_MODE);
+                                    if (jobStatusChangedListeners.size() > 0) {
+                                        jobStatusChangedListeners.forEach(
+                                                listener ->
+                                                        listener.onEvent(
+                                                                new DefaultJobCreatedEvent(
+                                                                        jobGraph.getJobID(),
+                                                                        jobGraph.getName(),
+                                                                        ((StreamGraph) pipeline)
+                                                                                .getLineageGraph(),
+                                                                        executionMode)));
+                                    }
+                                }
+                                clusterClient.close();
```

Comment:
Should this be in a finally block in case the listeners throws an error?

Reply:
I see this conversation is resolved - but I did not see an answer to what looks like a valid concern